bundles/config/org.eclipse.smarthome.config.core/src/main/java/org/eclipse/smarthome/config/core/BundleProcessorVetoManager.java

```diff
@@ -0,0 +1,146 @@
+/**
+ * Copyright (c) 2014-2016 by the respective copyright holders.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.smarthome.config.core;
+
+import java.security.AccessController;
+import java.security.PrivilegedAction;
+import java.util.HashSet;
+import java.util.Set;
+
+import org.eclipse.smarthome.config.core.BundleProcessor.BundleProcessorListener;
+import org.osgi.framework.Bundle;
+import org.osgi.framework.BundleReference;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.collect.LinkedListMultimap;
+import com.google.common.collect.Multimap;
+import com.google.common.collect.Multimaps;
+
+/**
+ * Utility class in order to delay an action until a binding's XML loading completed.
+ *
+ * It takes an action and whenever it is instructed to execute this action, it is going to check first if there are any
+ * vetoes due to the fact that the XML meta-data of the relevant binding might not be fully loaded. If this is the case,
+ * then the action will be delayed until the loading was finished.
+ *
+ * As this class is not an OSGi service itself, it needs some help from the outside by injecting/removing
+ * BundleProcessors via {@link #addBundleProcessor(BundleProcessor)} and
+ * {@link #removeBundleProcessor(BundleProcessor)}.
+ *
+ * @author Simon Kaufmann - Initial contribution and API
+ */
+public class BundleProcessorVetoManager<T> implements BundleProcessorListener {
+
+    public static interface Action<T> {
```

Comment:
If we want to "avoid modifiers which are implied by the context", you can drop "static" here.